### DISCUSSION TITLE
Alert (ding/dong) when stations join/drop

### DISCUSSION THREAD

**Discussion by AESilky** [MorseKOB/PyKOB#202] [Ideas]
I think it might be nice to have a background indication that a station joined a wire or dropped off.

When a station joins a wire you are currently connected to a 'ding' could sound. If a station that has been connected drops off a 'dong' could sound.

Both the 'join' and 'drop' notifications should be able to be enabled/disabled via configuration. A plus would be to configure the actual sounds (not just 'ding'/'dong').

Would anyone else find this to be a useful feature, or would you just tend to turn it off?

-Ed (ES)

**Comment by AESilky** [comment]
@leskerr , @pwdirks , @wamorganjr, @jchausler  - I pose this for two reasons...
1. To test the new GitHub Discussion feature that we were asked to try
2. As an actual idea that I though of while on Wire-11 the other day

**Comment by wamorganjr** [comment]
As long as there is a way to disable it. It's not something I would want at a museum or demo. 

73,

Chip N3IW 


-------- Original Message --------

**Comment by leskerr** [comment]
I would use the ding/dong feature.

**Comment by jchausler** [comment]
Hi Ed,I for one would turn it off, it's not authentic.73, Chris-----Original Message-----
From: Ed Silky 
Sent: Dec 9, 2020 5:15 AM
To: MorseKOB/PyKOB 
Cc: Subscribed 
Subject: [MorseKOB/PyKOB] Alert (ding/dong) when stations join/drop (#202)


I think it might be nice to have a background indication that a station joined a wire or dropped off.
When a station joins a wire you are currently connected to a 'ding' could sound. If a station that has been connected drops off a 'dong' could sound.
Both the 'join' and 'drop' sounds should be able to be enabled via configuration. A plus would be to configure the sounds.
Would anyone else find this to be a useful feature, or would you just tend to turn it off?
-Ed (ES)

**Reply by leskerr** [reply]
>Hi Ed, I for one would turn it off, it's not authentic.73, Chris

Neither is the station list. :-)

This is slightly off topic, but I wish there was a way to gray out the 'lurkers' in the station list. These are stations, like FB, that are always connected to wire 11 but seldom participate in the discussion. Once a station has transmitted, or if they connect to a session already in progress, their entry in the station list would then be presented in normal font.

**Comment by jchausler** [comment]
Hi Les,Possibly when folks first connect to a wire or are already on a wire but have not (recently) opened their key, they would be grayed out but once they open their key they would be displayed in a bolder font to show that they are active.  I wouldn't want the lurkers to go away completely, sometimes Ed Trump suddenly responds :-)As to the authenticity of the station list, in the real world of the telegraph the stations on a wire were fixed based on the physical routing of the wire.  Our "wires" span the world and the "routing" varies from session to session so the station list is to tell us where our wire is routed today :-)73, Chris-----Original Message-----

**Reply by leskerr** [reply]
>As to the authenticity of the station list, in the real world of the telegraph the stations on a wire were fixed based on the physical routing of the wire.  Our "wires" span the world and the "routing" varies from session to session so the station list is to tell us where our wire is routed today :-)

Good point! You got me. :-)

**Comment by AESilky** [comment]
The thing to remember is that the station list is local to the MKOB instance running. When you first connect you are the only station in the list. It is only as 'pings' are received or a station sends that that station is added to the list. There is a difference between a ping and a transmission - that is how MKOB knows to move a station to the bottom of the list when that station is sending and add a station just above the bottom (the current sender) the first time a ping is received. MKOB4 also keeps track of when a station last sent something, so it can tell the difference between stations that haven't sent anything (since this copy of MKOB connected to the wire), and how long it's been since a station (that has sent something) last sent (again, just since this copy of MKOB connected to the wire). However, it is possible that a number of stations could have been actively sending prior to this copy of MKOB connecting and just haven't sent anything since this copy of MKOB connected. In that case, those stations will appear as 'lurkers'.

**Comment by AESilky** [comment]
As long as it's understood that the highlighting is relative to the time that this MKOB4 has been connected to the wire - it wouldn't be to hard to make stations that have sent something be bold, while those that are connected but haven't yet sent (as far as this MKOB4 knows) be normal font.
